pandas/core/arrays/datetimelike.py

```diff
@@ -832,8 +1116,8 @@ def _time_shift(self, periods, freq=None):
                 freq = frequencies.to_offset(freq)
             offset = periods * freq
             result = self + offset
-            if hasattr(self, 'tz'):
-                result._tz = self.tz
+            if getattr(self, 'tz', None):
```

Comment:
why did this change?

Reply:
Reverting.

Reply:
Ahh never mind.

A tz-naive DatetimeIndex will still have a `.tz` attribute (like on master). It's value will be `None` though. We can't pass `tz=None` to `DatetimeTZDtype`.